src/_pytest/nodes.py

```diff
@@ -432,10 +444,39 @@ def __init__(
 
         super().__init__(name, parent, config, session, nodeid=nodeid, fspath=fspath)
 
+        self._norecursepatterns = self.config.getini("norecursedirs")
+
     @classmethod
     def from_parent(cls, parent, *, fspath):
         return cls._create(parent=parent, fspath=fspath)
 
+    def gethookproxy(self, fspath):
```

Comment:
can we try to keep this as a private api, we need to cahnge the path type eventually to oathlib

Reply:
Sure.  I've not checked, but how became it public now?
(I've thought that should not have changed)

Reply:
it was only on session, now its on every fs collector, so its  more public

Reply:
Oh, I see.  (it was on Package though also)
I'd say to prefix it with `_` then, hiding it - and if necessary keep a stub on the old classes?

Reply:
unless there is a known reason to keep the stubs, we might want to deprecate them while we are at it (so they can  be dropped after a while

Reply:
@RonnyPfannschmidt 
TBH I cannot see why having it "more public" is worse than "public".
So I'd rather do the deprecation when it is necessary, given that we would have to do it now already anyway, plus having a proxy method (i.e. more complications).

Reply:
its a api using the unmaintained  py.path.local api that we want to free pytest from eventually

Reply:
Sure.  But having it on 2 different classes seems worse to me than having it one base class.

Reply:
trough inheritance its now an api on Modules, doctest files and more

Reply:
@RonnyPfannschmidt 
And adding a deprecation in one place is easier than in two.

Can you say why "more public" is worse than "public"?
Having it in a central place allows for easier deprecation (one place).

Anyway, I was about to add a TypeError for when self is not Session/Package, but it is not trivial anymore to instantiate a Node.  How should a unittest look for this?  (i.e. how to create a Node - given the `from_parent` api)

Reply:
@blueyed good point about the deprecation warning, moving the method up while deprecating it is actually better, thanks for bringing it up

without deprecation its worse, with deprecation its actually  a reasonable tactical  move

as for the from_parent issue, i think i need to provide a MockParent for notes/test execution so we have something for unittests 

i can search for the code example to use in the meantime tomorrow, please hold me accountable if i miss that

Reply:
@RonnyPfannschmidt ping.

Reply:
Changed to AttributeError, and added a test.
Rebased on master to get unblocked due to `Node.from_parent`.